### PR TITLE
[chat] use official transformers and fix some issues

### LINKED ISSUE #4116: [chat] use official transformers and fix some errors
## Description

As reported in https://github.com/hpcaitech/ColossalAI/issues/4023, some use cases of `train_prompts` would fail.

We would like to adopt the following methods to try to fix them.

## TODOs

- [x] Upgrade the version of `transformers` that `coati` depends on to the latest version (i.e., official version).
- [x] Fix errors of `llama` model.
- [x] Remove `deberta` and `roberta` support as less used and still buggy.

### PR DESCRIPTION
**Pull request by CWHer**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

Close #4116 & #3639, #3389 & #3123, #3192.

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Comment by CWHer** [comment]
⚠️ This PR is not done yet.

The effect of removing `prepare_llama_tokenizer_and_embedding` hasn't been carefully examined.

**Comment by CWHer** [comment]
Commit 7f7ecba solves #3123.

File: .github/workflows/run_chatgpt_examples.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
       - name: Install Transformers
         run: |
           cd applications/Chat
-          git clone https://github.com/hpcaitech/transformers
+          git clone --depth 1 --branch v4.30.2 https://github.com/huggingface/transformers
```
Comment (ver217) [inline .github/workflows/run_chatgpt_examples.yml:42]: Is v4.30.2 a stable version? Can we install it from PYPI?

File: applications/Chat/examples/community/peft/train_peft_sft.py [hunk 2]
```diff
@@ -65,34 +64,31 @@ def train(args):
...
-            for name, param in model.named_parameters():
-                if not isinstance(param, ColoParameter):
-                    sub_module_name = '.'.join(name.split('.')[:-1])
-                    weight_name = name.split('.')[-1]
-                    sub_module = model.get_submodule(sub_module_name)
-                    setattr(sub_module, weight_name, ColoParameter(param))
-    else:
-        tokenizer.pad_token = tokenizer.eos_token
+
+    if args.model == 'llama' and args.strategy == 'colossalai_gemini':
```
Comment (ver217) [inline applications/Chat/examples/community/peft/train_peft_sft.py:83]: Resizing is not needed now, right?

File: applications/Chat/examples/train_prompts.py [hunk 3]
```diff
@@ -117,23 +107,20 @@ def main(args):
...
     elif args.model == 'bloom':
         tokenizer = BloomTokenizerFast.from_pretrained('bigscience/bloom-560m')
+        tokenizer.pad_token = tokenizer.eos_token
     elif args.model == 'opt':
         tokenizer = AutoTokenizer.from_pretrained("facebook/opt-350m")
+        tokenizer.pad_token = tokenizer.eos_token
     elif args.model == 'llama':
-        tokenizer = LlamaTokenizer.from_pretrained(args.pretrain)
+        tokenizer = LlamaTokenizer.from_pretrained("hf-internal-testing/llama-tokenizer")
         tokenizer.eos_token = '<\s>'
```
Comment (ver217) [inline applications/Chat/examples/train_prompts.py:119]: This is not needed

File: applications/Chat/examples/train_sft.py [hunk 4]
```diff
@@ -68,31 +67,27 @@ def train(args):
...
         tokenizer = AutoTokenizer.from_pretrained("facebook/opt-350m")
+        tokenizer.pad_token = tokenizer.eos_token
     elif args.model == 'llama':
         tokenizer = AutoTokenizer.from_pretrained(
-            args.pretrain,
+            "hf-internal-testing/llama-tokenizer",
             padding_side="right",
             use_fast=False,
         )
         tokenizer.eos_token = '<\s>'
```
Comment (ver217) [inline applications/Chat/examples/train_sft.py:77]: This is useless

File: .github/workflows/run_chatgpt_examples.yml (hunk 1 above)
Comment (CWHer) [reply .github/workflows/run_chatgpt_examples.yml]: This is fixed in 3d5d4ec.

File: applications/Chat/examples/train_sft.py (hunk 4 above)
Comment (CWHer) [reply applications/Chat/examples/train_sft.py:77]: Fixed in 2ceba11.

File: applications/Chat/examples/train_prompts.py (hunk 3 above)
Comment (CWHer) [reply applications/Chat/examples/train_prompts.py:119]: Fixed in 2ceba11.

File: applications/Chat/examples/community/peft/train_peft_sft.py (hunk 2 above)
Comment (CWHer) [reply applications/Chat/examples/community/peft/train_peft_sft.py:83]: I have sent an email to the author.

**Review by ver217 — APPROVED** [review]

**Pull request merged**